ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -466,7 +787,153 @@ public void replicate() {
             time.milliseconds() - fastestReplicaFinishTime, replicatingFromRemoteColo, datacenterName);
       }
     }
-    maybeSleepAfterReplication(remoteReplicaGroups.isEmpty());
+    maybeSleepAfterReplication(remoteReplicaGroups.isEmpty(), 1);
+  }
+
+  /**
+   * Do replication for replicas grouped by {@link DataNodeId}
+   * We will keep trying to generate groups until we do not get any groups by calling generateRemoteReplicaGroups.
+   * We will generate requests and then process responses when received for these groups.
+   *
+   * A replication cycle between two replicas involves the following steps:
+   *    1. Exchange metadata : fetch the metadata of blobs added to remote replica since the last synchronization point
+   *    and filter the ones missing in local store.
+   *    2. Fetch missing blobs: fetch the missing blobs by issuing GET request to remote replica and write them to
+   *       the local store
+   *
+   *  During cross-colo replication, depending on the {@link ReplicationModelType}, the missing blobs are either fetched
+   *  from all remote replicas (if modelType == ALL_TO_ALL) or only fetched for local leader replicas from their remote
+   *  leader replicas (if modelType == LEADER_BASED). In the latter case, non-leader replica pairs (leader <-> standby,
+   *  standby <-> leader, standby <-> standby) will get their missing blobs from their corresponding leader<->leader
+   *  exchanges and intra-dc replication.
+   *
+   *  Here is a table listing on what is exchanged between local and remote replicas based on their roles
+   *  (leader/standby) when {@link ReplicationModelType is LEADER_BASED}.
+   *
+   *              |   Local Leader    |     Local Standby   |   Remote Leader   |  Remote Standby
+   *            -------------------------------------------------------------------------------------
+   *     Leader:  |        ---        |  metadata and data  | metadata and data |   metadata only
+   *     Standby: | metadata and data |  metadata and data  | metadata only     |   metadata only
+   *
+   */
+  public void replicateLimitedAcyclic() {
+    long oneRoundStartTimeMs = time.milliseconds();
+    shouldTerminateCurrentCycle = false;
+
+    logger.trace("Thread name: {} Start RemoteReplicaGroup replication", threadName);
```

Comment:
rm "Thread name", the logger prints that anyways. Just put your msg in the logger

Reply:
In this code, everywhere it is written like this.
also in test logs, we need to print manually.